flexmeasures/api/dev/sensor_data.py

```diff
@@ -0,0 +1,26 @@
+from webargs.flaskparser import use_args
+
+from flexmeasures.data.config import db
+from flexmeasures.data.models.time_series import TimedBelief
+from flexmeasures.api.common.schemas.sensor_data import SensorDataSchema
+
+
+@use_args(
+    SensorDataSchema(),
+    location="json",
+)
+def post_data(sensor_data):
+    """POST to /sensorData
+
+    Experimental dev feature which uses timely-beliefs
+    to create and save the data structure.
+    """
+    beliefs = SensorDataSchema.load_bdf(sensor_data)
```

Comment:
I tried that. It's not a good fit.

Marshmallow first de-serializes (turning JSON values into the fields we defined), then validates (schema-level first, field-level second). Your wish would require us to fit all of that into our _deserialize function before we turn the content into a BeliefsDataFrame, which it is not meant to be, and I'm not sure how I'd fight Marshmallow to accomplish it.

I believe this is cleaner. I also think you'd "love" this for aesthetic reasons, rather than clear practical reasons.

Reply:
Agreed!